File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        if(n == 1 || n == 0){ return 0;  }
        int j = 1;
        int profit = 0;
        int min = Integer.MAX_VALUE;
        for(int i = 0;i<n && j<n;i++){
            if(prices[i]<min){
                min = prices[i];
            }else if(prices[i]-min>profit){ 
                profit = prices[i]-min;
            }
        }
        return profit;
    }
}

File: LeetCode/Easy/0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProfit = 0
        minm = prices[0]
        for i in range(1,len(prices)):
            minm = min(prices[i],minm)
            if prices[i] - minm>maxProfit:
                maxProfit = prices[i]-minm
        return maxProfit



